tmk_core/protocol/usb_hid/override_wiring.c

```diff
@@ -2,7 +2,13 @@
  * To keep Timer0 for common/timer.c override arduino/wiring.c.
  */
 #define __DELAY_BACKWARD_COMPATIBLE__
+
+// Need to disable GCC's "maybe-uninitialized" warning for this file, as it causes issues when running `KEEP_INTERMEDIATES=yes`.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
 #include <util/delay.h>
```

Comment:
I think this could just be changed to `wait.h` (+ the `_delay_*` calls below).

Reply:
Will stage a new PR to convert all usages of `util/delay.h` to `wait.h` after this one.
In this PR though, I have moved the `#pragma` to `wait.h` instead, and swapped over this occurrence.

Reply:
e.g. [`74a8e2d` (#20339)](https://github.com/qmk/qmk_firmware/pull/20339/commits/74a8e2d88f73b69d5964a2667d8c13bcb6bfc7d6)